### PR TITLE
Bring intermediate CA setup into the app 

### PR DESCRIPTION
**Pull request by suprjinx**
Still expecting a pre-configured root, referenced by config

### CODE REVIEW

File: app/lib/clients/vault.rb [hunk 1]
```diff
@@ -13,7 +14,9 @@ def kv_read(path)
       end
 
       def kv_write(path, data)
-        enable_engine(kv_mount, "kv-v2")
+        unless client.sys.mounts.key?(kv_mount.to_sym)
+          enable_engine(kv_mount, "kv-v2")
```
Comment (dave-gantenbein) [inline app/lib/clients/vault.rb:18]: Can you extract the magic string here and below as you did for kv_mount?

File: app/lib/clients/vault.rb [hunk 2]
```diff
@@ -25,23 +28,89 @@ def kv_delete(path)
...
 
+      def enable_ca
+        # if mount exists, assume configuration is done
+        if client.sys.mounts.key?(intermediate_ca_mount.to_sym)
+          return
+        end
 
-      def kv_mount
-        Rails.configuration.astral[:vault_kv_mount]
+        # create the mount
```
Comment (dave-gantenbein) [inline app/lib/clients/vault.rb:70]: Can we test this in some way? It does a bunch of stuff here!

File: app/lib/clients/vault.rb
Comment (dave-gantenbein) [inline app/lib/clients/vault.rb:1]: Are you still sure you want to have cert and kv operations in this same class? Asking for my friend SRP....

Comment (suprjinx) [reply app/lib/clients/vault.rb:1]: well i moved these into a single client class but with separate services defining the interface. It's all "talking to Vault" and they share a lot of stuff, like "engine mount", etc.

File: app/lib/clients/vault.rb (hunk 1 above)
Comment (suprjinx) [reply app/lib/clients/vault.rb]: done

File: app/lib/clients/vault.rb
Comment (suprjinx) [reply app/lib/clients/vault.rb:1]: split into separate files reopening the class, let me know what you think

File: app/lib/clients/vault.rb (hunk 2 above)
Comment (suprjinx) [reply app/lib/clients/vault.rb]: it's implicitly tested by the integration tests but I'll work on adding more

Comment (suprjinx) [reply app/lib/clients/vault.rb]: tests added for the 'configure_pki' and 'configure_kv'

File: app/lib/clients/vault.rb
Comment (dave-gantenbein) [reply app/lib/clients/vault.rb:1]: Beautiful! You'll thank me later :)

**Review by dave-gantenbein — APPROVED** [review]

**Pull request merged**